src/containers/Tenant/QueryEditor/SavedQueries/SavedQueries.tsx

```diff
@@ -0,0 +1,151 @@
+                {i18n('delete-dialog.question')}
+                <span className={b('dialog-query-name')}>{` ${queryName}?`}</span>
+            </Dialog.Body>
+            <Dialog.Footer
+                textButtonApply={i18n('delete-dialog.delete')}
+                textButtonCancel={i18n('delete-dialog.cancel')}
+                onClickButtonCancel={onCancelClick}
+                onClickButtonApply={onConfirmClick}
+            />
+        </Dialog>
+    );
+};
+
+interface SavedQueriesProps {
+    savedQueries: SavedQuery[];
+    changeUserInput: (value: {input: string}) => void;
+    onDeleteQuery: (queryName: string) => void;
+}
+
+export const SavedQueries = ({savedQueries, changeUserInput, onDeleteQuery}: SavedQueriesProps) => {
+    const dispatch = useDispatch();
+
+    const [isDeleteDialogVisible, setIsDeleteDialogVisible] = useState(false);
+    const [queryNameToDelete, setQueryNameToDelete] = useState<string>('');
+
+    const closeDeleteDialog = () => {
+        setIsDeleteDialogVisible(false);
+        setQueryNameToDelete('');
+    };
+
+    const onCancelDeleteClick = () => {
+        closeDeleteDialog();
+    };
+
+    const onConfirmDeleteClick = () => {
+        closeDeleteDialog();
+        onDeleteQuery(queryNameToDelete);
+        setQueryNameToDelete('');
+    };
+
+    const onQueryClick = (queryText: string, queryName: string) => {
+        changeUserInput({input: queryText});
+        dispatch(setQueryNameToEdit(queryName));
+        dispatch(setQueryTab(TENANT_QUERY_TABS_ID.newQuery));
+    };
+
+    const onDeleteQueryClick = (queryName: string) => {
+        return (event: MouseEvent) => {
+            event.stopPropagation();
+            setIsDeleteDialogVisible(true);
+            setQueryNameToDelete(queryName);
+        };
+    };
+
+    const columns: Column<SavedQuery>[] = [
+        {
+            name: 'name',
+            header: 'Name',
+            render: ({row: query}) => <div className={b('query-name')}>{query.name}</div>,
+            width: 200,
+        },
+        {
+            name: 'body',
+            header: 'Query Text',
+            render: ({row: query}) => (
+                <div className={b('query')}>
+                    <div className={b('query-body')}>
+                        <TruncatedQuery value={query.body} maxQueryHeight={MAX_QUERY_HEIGHT} />
+                    </div>
+                    <span className={b('control-delete')} onClick={onDeleteQueryClick(query.name)}>
+                        <IconWrapper
+                            name="trash"
+                            viewBox="0 0 24 24"
+                            height={14}
+                            width={14}
+                            className={b('control-button')}
+                        />
+                    </span>
+                </div>
+            ),
+            sortable: false,
+        },
+    ];
+
+    return (
+        <>
+            <div className={b()}>
+                <DataTable
```

Comment:
It seems that search by name or by query text would be very helpful here and in `History` as well.

Reply:
We'll discuss it on call. It's not the issue of this PR